storage/wdpost_run.go

```diff
@@ -759,6 +759,18 @@ func (s *WindowPoStScheduler) setSender(ctx context.Context, msg *types.Message,
 		msg.From = s.worker
 		return
 	}
+	worker, err := s.api.StateAccountKey(ctx, mi.Worker, types.EmptyTSK)
+	if err != nil {
+		log.Errorw("error getting account key", "error", err)
+
+		msg.From = s.worker
+		return
+	}
+	// worker should keep sync with state on chain, if not, will met error when estimate msg gas.
+	if s.worker != worker {
```

Comment:
If we error above, `worker`, this condition will be true, and we'll set undefined worker address

Reply:
If error above, it will return the func in the `if` block, so it will never change the worker to undefined

Reply:
Ah, I see